loki/transformations/pool_allocator.py

```diff
@@ -222,7 +223,7 @@ def transform_subroutine(self, routine, **kwargs):
         ignore = item.ignore if item else ()
         targets = as_tuple(kwargs.get('targets', None))
 
-        self.stack_type_kind = 'JPRB'
+        self.stack_type_kind = 'REAL64'
         if item:
             if (real_kind := item.config.get('real_kind', None)):
                 self.stack_type_kind = real_kind
```

Comment:
You might consider just dropping this property and directly writing REAL64 everywhere. This kind is now baked in explicitly due to the 8 byte alignment that there's not really any value in having this parameterised